browser/src/Editor/SimpleNeovimEditor.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * NeovimEditor.ts
+ *
+ * IEditor implementation for Neovim
+ */
+
+import * as React from "react"
+
+import { IncrementalDeltaRegionTracker } from "./../DeltaRegionTracker"
+import { NeovimInstance } from "./../neovim"
+import { CanvasRenderer, DOMRenderer, INeovimRenderer } from "./../Renderer"
+import { NeovimScreen } from "./../Screen"
+
+import * as Config from "./../Config"
+
+// import { PluginManager } from "./../Plugins/PluginManager"
+
+import { Keyboard } from "./../Input/Keyboard"
+import { IEditor } from "./Editor"
+
+import { NeovimRenderer } from "./NeovimRenderer"
+
+export class DummyPluginManager {
+    public getAllRuntimePaths(): string[] {
+        return []
+    }
+
+    public startPlugins(neovimInstance: NeovimInstance): void {
+        // no-op
+    }
+}
+
+export class SimpleNeovimEditor implements IEditor {
+
+    private _neovimInstance: NeovimInstance
+    private _deltaRegionManager: IncrementalDeltaRegionTracker
+    private _renderer: INeovimRenderer
+    private _screen: NeovimScreen
+
+    private _pendingTimeout: any = null
+    private _pendingAnimationFrame: boolean = false
+
+    constructor(
+        private _config: Config.Config = Config.instance(),
+    ) {
+        // TODO: How to get rid of this?
+        this._neovimInstance = new NeovimInstance(new DummyPluginManager(), 100, 100)
+        this._neovimInstance.setInitVim("C:/oni/test.vim")
+        this._deltaRegionManager = new IncrementalDeltaRegionTracker()
+        this._screen = new NeovimScreen(this._deltaRegionManager)
+
+        this._renderer = this._config.getValue("editor.renderer") === "canvas" ? new CanvasRenderer() : new DOMRenderer()
+
+        this._render()
+
+        this._config.registerListener(() => this._onConfigChanged())
+
+        this._neovimInstance.on("action", (action: any) => {
+            this._renderer.onAction(action)
+            this._screen.dispatch(action)
+
+            this._scheduleRender()
+
+            if (!this._pendingTimeout) {
+                this._pendingTimeout = setTimeout(() => this._onUpdate(), 0)
+            }
+        })
+
+        const keyboard = new Keyboard()
```

Comment:
Yes, good points!

> above with the other editors, we can have on load event with what kind of editor abstraction it is.

This is interesting, and something I'm still trying to reason about - there are specific things we can do with a neovim-editor (like send it normal mode commands, execute VimL, etc), but how do plugins reason about this vs a non-neovim-backed editor? I like the idea of having a load event or some way to know the capabilities that the editor has.

> Thanks for listening to my ideas, Oni just keeps more and more in line with what I was thinking for an editor.

Great to hear :)